lib/internal/process/warning.js

```diff
@@ -17,10 +17,14 @@ function lazyOption() {
   return warningFile;
 }
 
+// If we can't write to stderr, we'd like to make this a noop,
+// so use console.error.
+let error;
 function writeOut(message) {
-  if (console && typeof console.error === 'function')
-    return console.error(message);
-  process._rawDebug(message);
+  if (!error) {
+    error = require('internal/console/global').error;
+  }
+  error(message);
```

Comment:
The question is should we support manipulating globals at all? If so I believe it has to be done explicitly through global.console with a comment that this intends to support that use case, instead of having console calls scattered around

Reply:
The same could be said for using the internal `Array` over the `global.Array`. I think we're far along enough this path already.

Reply:
@guybedford We are now migrating towards the primordials, actually..(I think it's possible to reach a point where we only allow explicit access to the global builtins with eslint-disable, if eslint does support banning access to global JS builtins..)